philipmeneghini/InstaConnect
# Search Bar Update

**Summary**
Successfully implemented a `useDebounce` hook which will only send a search request once the user has not pressed a new key in a couple seconds. The pop up menu also contains helpful _No Search Results_ and _Please type your search_ messages. When clicking out of focus the results menu will disappear. It will only reappear when clicking back into the search bar and typing or hitting the enter button

![searchupdate](https://github.com/philipmeneghini/InstaConnect/assets/63440066/38115a54-9389-434b-b9c2-76a722589a87)

**Testing**

- Make sure all the correct search results populate in pop up
- Test that the `useDebounce` works by tracking calls to the backend when typing lots of letters
- Test that the proper message gets displayed when nothing is passed into the search or there are no results
- Check for all general functionality
